e2e/ctl_v3_make_mirror_test.go

```diff
@@ -25,106 +25,31 @@ func TestCtlV3MakeMirrorModifyDestPrefix(t *testing.T) { testCtl(t, makeMirrorMo
-	cfg := configAutoTLS
-	cfg.clusterSize = 1
-	cfg.basePort = 10000
-	cx2 := ctlCtx{
-		t:           cx.t,
-		cfg:         cfg,
-		dialTimeout: 7 * time.Second,
-	}
-
-	epc, err := newEtcdProcessCluster(&cx2.cfg)
-	if err != nil {
-		cx.t.Fatalf("could not start etcd process cluster (%v)", err)
-	}
-	cx2.epc = epc
-
-	defer func() {
-		if err = cx2.epc.Close(); err != nil {
-			cx.t.Fatalf("error closing etcd processes (%v)", err)
-		}
-	}()
-
-	cmdArgs := append(cx.PrefixArgs(), "make-mirror", fmt.Sprintf("localhost:%d", cfg.basePort))
-	proc, err := spawnCmd(cmdArgs)
-	if err != nil {
-		cx.t.Fatal(err)
-	}
-	defer func() {
-		err = proc.Stop()
-		if err != nil {
-			cx.t.Fatal(err)
-		}
-	}()
-
+	flags := []string{}
 	var kvs = []kv{{"key1", "val1"}, {"key2", "val2"}, {"key3", "val3"}}
-	for i := range kvs {
-		if err = ctlV3Put(cx, kvs[i].key, kvs[i].val, ""); err != nil {
-			cx.t.Fatal(err)
-		}
-	}
-	if err = ctlV3Get(cx, []string{"key", "--prefix"}, kvs...); err != nil {
-		cx.t.Fatal(err)
-	}
-	if err = ctlV3Watch(cx2, []string{"key", "--rev", "1", "--prefix"}, kvs...); err != nil {
-		cx.t.Fatal(err)
-	}
+	var prefix = "key"
+	testMirrorOp(cx, flags, kvs, kvs, prefix, prefix)
 }
 
 func makeMirrorModifyDestPrefixTest(cx ctlCtx) {
-	// set up another cluster to mirror with
-	mirrorcfg := configAutoTLS
-	mirrorcfg.clusterSize = 1
-	mirrorcfg.basePort = 10000
-	mirrorctx := ctlCtx{
-		t:           cx.t,
-		cfg:         mirrorcfg,
-		dialTimeout: 7 * time.Second,
-	}
-
-	mirrorepc, err := newEtcdProcessCluster(&mirrorctx.cfg)
-	if err != nil {
-		cx.t.Fatalf("could not start etcd process cluster (%v)", err)
-	}
-	mirrorctx.epc = mirrorepc
-
-	defer func() {
-		if err = mirrorctx.epc.Close(); err != nil {
-			cx.t.Fatalf("error closing etcd processes (%v)", err)
-		}
-	}()
-
-	cmdArgs := append(cx.PrefixArgs(), "make-mirror", "--prefix", "o_", "--dest-prefix", "d_", fmt.Sprintf("localhost:%d", mirrorcfg.basePort))
-	proc, err := spawnCmd(cmdArgs)
-	if err != nil {
-		cx.t.Fatal(err)
-	}
-	defer func() {
-		err = proc.Stop()
-		if err != nil {
-			cx.t.Fatal(err)
-		}
-	}()
-
+	flags := []string{"--prefix", "o_", "--dest-prefix", "d_"}
 	var kvs = []kv{{"o_key1", "val1"}, {"o_key2", "val2"}, {"o_key3", "val3"}}
-	for i := range kvs {
-		if err = ctlV3Put(cx, kvs[i].key, kvs[i].val, ""); err != nil {
-			cx.t.Fatal(err)
-		}
-	}
-	if err = ctlV3Get(cx, []string{"o_", "--prefix"}, kvs...); err != nil {
-		cx.t.Fatal(err)
-	}
-
 	var kvs2 = []kv{{"d_key1", "val1"}, {"d_key2", "val2"}, {"d_key3", "val3"}}
-	if err = ctlV3Watch(mirrorctx, []string{"d_", "--rev", "1", "--prefix"}, kvs2...); err != nil {
-		cx.t.Fatal(err)
-	}
+	var srcprefix = "o_"
+	var destprefix = "d_"
+	testMirrorOp(cx, flags, kvs, kvs2, srcprefix, destprefix)
 }
 
 func makeMirrorNoDestPrefixTest(cx ctlCtx) {
+	flags := []string{"--prefix", "o_", "--no-dest-prefix"}
+	var kvs = []kv{{"o_key1", "val1"}, {"o_key2", "val2"}, {"o_key3", "val3"}}
```

Comment:
@xiang90 sure will do the changes. Just one confirmation required, you want me to move all the variables under var block. right? and you want this to be done for all the functions right?